include/revng/ADT/SortedVector.h

```diff
@@ -396,19 +417,16 @@ class SortedVector {
     return not compareKeys(LHS, RHS) and not compareKeys(RHS, LHS);
   }
 
-  template<bool KeepFirst>
+  template<bool EnsureUnique>
   void sort() {
-    // Sort
-    std::stable_sort(begin(), end(), compareElements);
-
-    // Remove duplicates keeping the last instance of each element
-    auto NewEnd = end();
-    if (KeepFirst) {
-      NewEnd = std::unique(begin(), end(), elementsEqual);
+    if constexpr (EnsureUnique) {
+      std::sort(begin(), end(), compareElements);
+      revng_check(std::adjacent_find(begin(), end(), elementsEqual) == end());
     } else {
-      NewEnd = unique_last(begin(), end(), elementsEqual);
+      std::stable_sort(begin(), end(), compareElements);
+      auto NewEnd = unique_last(begin(), end(), elementsEqual);
+      TheVector.erase(NewEnd, end());
     }
```

Comment:
Do not link commits. Say something like:

> Please review the commit named "SortedVector: improve new element insertion"

Reply:
There's only one commit touching `SortedVector`.

In short, all the changes to the file by this PR.

Reply:
> it also makes the non-assign version of the inserter a lot safer by explicitly asserting the fact that no duplicates were
inserted while it was in use

I understand this is useful but deviates from the `std` semantics of `insert`.
I'm not against this in principle, but if you want to do this, stress this in the doxygen of the method.

Also, consider adding some tests to `KeyedObjectsContainers.cpp` for testing new behaviors, but I'm afraid what said above will make it hard since testing tries to ensure that `SortedVector` behaves like other `std` data structures in this sense.